javalanglib/src/main/scala/java/lang/ClassValue.scala

```diff
@@ -0,0 +1,99 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package java.lang
+
+import java.util.HashMap
+
+import scala.scalajs.js
+import scala.scalajs.runtime.linkingInfo
+
+abstract class ClassValue[T] protected () {
+  import ClassValue._
+
+  private val jsMap: js.Dynamic = {
+    if (linkingInfo.assumingES6 || js.typeOf(js.Dynamic.global.Map) != "undefined")
+      js.Dynamic.newInstance(js.Dynamic.global.Map)()
+    else
+      null
+  }
+
+  @inline
+  private def useJSMap: scala.Boolean = {
+    /* The linking-info test allows to constant-fold this method as `true` when
+     * emitting ES 2015 code, which allows to dead-code-eliminate the branches
+     * using `HashMap`s, and therefore `HashMap` itself.
+     */
+    linkingInfo.assumingES6 || jsMap != null
+  }
+
+  /* We use a HashMap instead of an IdentityHashMap because the latter is
+   * implemented in terms of the former anyway, to a HashMap is leaner and
+   * faster.
+   */
+  private val javaMap: HashMap[Class[_], Any] =
+    if (useJSMap) null
+    else new HashMap()
+
+  protected def computeValue(`type`: Class[_]): T
+
+  def get(`type`: Class[T]): T = {
```

Comment:
IIUC, this signature is wrong, it should be `Class[_]`.

Reply:
Oops, good catch. Fixed.